libraries/core_libs/network/include/network/tarcap/shared_states/pbft_syncing_state.hpp

```diff
@@ -73,13 +98,34 @@ class PbftSyncingState {
   // Number of seconds needed for ongoing syncing to be declared as inactive
   static constexpr std::chrono::seconds kSyncingInactivityThreshold{60};
 
+  // Number of milliseconds for peer asking sync period timeout
+  // corner case restart/rebuild DB may ask previous periods again
+  const std::chrono::milliseconds kPeerAskingPeriodTimeout{1000};
+
   // What time was received last syncing packet
   std::chrono::steady_clock::time_point last_received_sync_packet_time_{std::chrono::steady_clock::now()};
   mutable std::shared_mutex time_mutex_;
 
   // Peer that the node is syncing with
   std::shared_ptr<TaraxaPeer> peer_;
   mutable std::shared_mutex peer_mutex_;
+
+  /**
+   * @brief Table for peer asking sync period
+   *        Table key: peer node ID, value: a pair
+   *        Pair key: peer last asking request period, pair value: peer last asking request time point
+   */
+  std::unordered_map<dev::p2p::NodeID, std::pair<uint64_t, std::chrono::steady_clock::time_point>>
+      asking_period_peers_table_;
+  mutable boost::shared_mutex asking_period_mutex_;
```

Comment:
I agree with Matus, even this could be part of TaraxaPeer class - if you mark some peer as malicious due to this functionality, he wont be able to reconnect anyway so why would you keep this in some separate table that you have to clean separately, etc... ?

Reply:
I put the tables in PbftSyncingState because I want to make pbft syncing components in the same module

Reply:
I strongly disagree. It is peer's property and it should be in Peer class + we would not need to create any additional containers that have to be cleaned etc...